commit 61a1b8ffb1c68912484c838c0392633d34716205
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Mon Jan 16 21:02:26 2017 +0000

    Revert "bcm2835-i2s: Changes for allowing asymmetric sample formats."
    
    This reverts commit f5a6236a32e82068122301d246a94ca755d61704.
    
    See: https://github.com/raspberrypi/linux/issues/1799
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/sound/soc/bcm/bcm2835-i2s.c b/sound/soc/bcm/bcm2835-i2s.c
index 171c2401dfe1..6ba20498202e 100644
--- a/sound/soc/bcm/bcm2835-i2s.c
+++ b/sound/soc/bcm/bcm2835-i2s.c
@@ -237,9 +237,7 @@ static int bcm2835_i2s_hw_params(struct snd_pcm_substream *substream,
 	unsigned int sampling_rate = params_rate(params);
 	unsigned int data_length, data_delay, bclk_ratio;
 	unsigned int ch1pos, ch2pos, mode, format;
-	unsigned int previous_ftxp, previous_frxp;
 	uint32_t csreg;
-	bool packed;
 
 	/*
 	 * If a stream is already enabled,
@@ -322,46 +320,26 @@ static int bcm2835_i2s_hw_params(struct snd_pcm_substream *substream,
 		return -EINVAL;
 	}
 
-	/* Set the format for the matching stream direction. */
-	switch (substream->stream) {
-	case SNDRV_PCM_STREAM_PLAYBACK:
-		regmap_write(dev->i2s_regmap, BCM2835_I2S_TXC_A_REG, format);
-		break;
-	case SNDRV_PCM_STREAM_CAPTURE:
-		regmap_write(dev->i2s_regmap, BCM2835_I2S_RXC_A_REG, format);
-		break;
-	default:
-		return -EINVAL;
-	}
+	/*
+	 * Set format for both streams.
+	 * We cannot set another frame length
+	 * (and therefore word length) anyway,
+	 * so the format will be the same.
+	 */
+	regmap_write(dev->i2s_regmap, BCM2835_I2S_RXC_A_REG, format);
+	regmap_write(dev->i2s_regmap, BCM2835_I2S_TXC_A_REG, format);
 
 	/* Setup the I2S mode */
-	/* Keep existing FTXP and FRXP values. */
-	regmap_read(dev->i2s_regmap, BCM2835_I2S_MODE_A_REG, &mode);
-
-	previous_ftxp = mode & BCM2835_I2S_FTXP;
-	previous_frxp = mode & BCM2835_I2S_FRXP;
-
 	mode = 0;
 
-	/*
-	 * Retain the frame packed mode (2 channels per 32 bit word)
-	 * of the other direction stream intact. The formats of each
-	 * direction can be different as long as the frame length is
-	 * shared for both.
-	 */
-	packed = data_length <= 16;
-
-	switch (substream->stream) {
-	case SNDRV_PCM_STREAM_PLAYBACK:
-		mode |= previous_frxp;
-		mode |= packed ? BCM2835_I2S_FTXP : 0;
-		break;
-	case SNDRV_PCM_STREAM_CAPTURE:
-		mode |= previous_ftxp;
-		mode |= packed ? BCM2835_I2S_FRXP : 0;
-		break;
-	default:
-		return -EINVAL;
+	if (data_length <= 16) {
+		/*
+		 * Use frame packed mode (2 channels per 32 bit word)
+		 * We cannot set another frame length in the second stream
+		 * (and therefore word length) anyway,
+		 * so the format will be the same.
+		 */
+		mode |= BCM2835_I2S_FTXP | BCM2835_I2S_FRXP;
 	}
 
 	mode |= BCM2835_I2S_FLEN(bclk_ratio - 1);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1131_61a1b8ffb1c68912484c838c0392633d34716205.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1131_61a1b8ffb1c68912484c838c0392633d34716205.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
